Yoast/wordpress-seo
# Implement readability analysis container

## Summary

This PR can be summarized in the following changelog entry:

* Added readability analysis to the sidebar.

## Relevant technical choices:

* Used a similar style as https://github.com/Yoast/wordpress-seo/pull/10450
* Renamed the old ReadabilityAnalysis to ReadabilityAnalysisLegacy, and still used it in AnalysisSection.

Note that I have used Collapsible for now. This because the issue creating the collapsibleHeaderWithIcon issue is still in progress. It is for this reason (and to prevent an additional PR for yoast-copmonents) that I'm still using the old way to import for now.
* Should that collapsibleHeaderWithIcon be merged, than this PR needs to be updated with that collapsible Header. 
* Should this PR be merged before the collapsibleHeaderWithIcon is merged, a separate issue needs to be made describing that this collapsible needs to be updated when the collapsibleHeaderWithIcon is merged.

The fact that the sidebar doesn't remember the collapsed state of the collapsible is not part of the scope of this issue. (When you close and open the sidebar, it doesn't remember whether the collapsible was open or not).

I also noticed the helpText for the marking buttons thread outside the bounds of the sidebar. This should be fixed in the future, but not now.

## Test instructions

This PR can be tested by following these steps:

* Checkout branch and link its sibling pr from yoast-components: https://github.com/Yoast/yoast-components/pull/675.
* Make sure the feature flag `define('YOAST_FEATURE_GUTENBERG_SIDEBAR', true );` is set to true.
* Visit a gutenberg edit-post page.
* Open the Yoast Plugin in the sidebar, and verify that the components there *APPROXIMATE* the look required in the issue. Styling will be finalized a later point.
* Verify that the Readability Results in the sidebar update when changing the content in the editor.

## Quality assurance


Fixes #10304
